### PR TITLE
Linear trend

### PR DESCRIPTION
**Pull request by wd60622**
## Description

Linear trend using change points

CC: @cetagostini 

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [ ] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pymc_marketing/mmm/linear_trend.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Linear trend using change points.
+
+Examples
+--------
+Define a linear trend with 8 changepoints:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/linear_trend.py:18]: Are the change points equally spaced?

File: pymc_marketing/mmm/linear_trend.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+        where :math:`M` is the number of change points.
+
+    The priors for the trend parameters are:
+
+    - :math:`k \sim \text{Normal}(0, 0.05)`
+    - :math:`\delta_m \sim \text{Laplace}(0, 0.25)`
+
+    Adapted from MBrouns/timeseers package:
+    https://github.com/MBrouns/timeseers/blob/master/src/timeseers/linear_trend.py
```
Comment (juanitorduz) [inline pymc_marketing/mmm/linear_trend.py:99]: Maybe we can add this into the doctrines as 

```python

References
-----------
Adapted from MBrouns/timeseers package:
    https://github.com/MBrouns/timeseers/blob/master/src/timeseers/linear_trend.py
```

File: pymc_marketing/mmm/linear_trend.py [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+        n_dates = 52 * n_years
+        first_date = "2020-01-01"
+        dates = pd.date_range(first_date, periods=n_dates, freq="W-MON")
+        dayofyear = dates.dayofyear.to_numpy()
+        t = (dates - dates[0]).days.to_numpy()
+        t = t / 365.25
+
+        coords = {
+            "date": dates,
+        }
```
Comment (juanitorduz) [inline pymc_marketing/mmm/linear_trend.py:145]: this can be a single line right?

File: pymc_marketing/mmm/linear_trend.py [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+
+    Examples
+    --------
+    Linear trend with 10 changepoints:
+
+    .. code-block:: python
+
+        from pymc_marketing.mmm import LinearTrend
+
+        trend = LinearTrend(n_changepoints=10)
```
Comment (juanitorduz) [inline pymc_marketing/mmm/linear_trend.py:123]: can we add an example where we pass the priors explicitly?

File: pymc_marketing/mmm/linear_trend.py [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+        include_intercept: bool = False,
+    ) -> None:
+        self.n_changepoints = n_changepoints
+        self.include_intercept = include_intercept
+        self.priors: dict[str, Prior] = priors or self.default_priors.copy()
+        self.dims: Dims = dims or ()
+
+        self._checks()
+
+    def _checks(self) -> None:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/linear_trend.py:170]: We could use a pydantic Base model or https://docs.pydantic.dev/latest/concepts/validation_decorator/ as in 

https://github.com/pymc-labs/pymc-marketing/blob/f2318aee2653142f39bf0465e7e6770060c731bb/pymc_marketing/mmm/mmm.py#L73-L74

Comment (juanitorduz) [reply pymc_marketing/mmm/linear_trend.py:170]: applies to all the validation steps below ;)

**Review by juanitorduz — COMMENTED** [review]
Looks great! I left some suggestions, let me know what you think.

**Comment by juanitorduz** [comment]
Also, latex need some work

image

File: pymc_marketing/mmm/linear_trend.py (hunk 5 above)
Comment (wd60622) [reply pymc_marketing/mmm/linear_trend.py:170]: Yeah, I will make that adjustment

**Review by wd60622 — COMMENTED** [review]
Thanks for the feedback, will work through implementation when I get the chance

File: pymc_marketing/mmm/linear_trend.py (hunk 1 above)
Comment (juanitorduz) [reply pymc_marketing/mmm/linear_trend.py:18]: yeah! I saw it later! All good!

File: pymc_marketing/mmm/linear_trend.py (hunk 3 above)
Comment (juanitorduz) [reply pymc_marketing/mmm/linear_trend.py:145]: No, I jiust meant:

```python
        coords = {"date": dates}
```
XD

File: pymc_marketing/mmm/linear_trend.py (hunk 4 above)
Comment (juanitorduz) [reply pymc_marketing/mmm/linear_trend.py:123]: maybe the class?

**Comment by wd60622** [comment]
Addressed all of the feedback @juanitorduz 
Waiting on the docs to build in order to check the new latex 🚀

**Review by juanitorduz — APPROVED** [review]
great!

**Pull request merged**